pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/basic/BooleanInstantiationRule.java

```diff
@@ -63,10 +64,20 @@ public Object visit(ASTAllocationExpression node, Object data) {
 	        if (node.hasDescendantOfType(ASTArrayDimsAndInits.class)) {
 	            return super.visit(node, data);
 	        }
-	        if (TypeHelper.isA((ASTClassOrInterfaceType) node.jjtGetChild(0), Boolean.class)) {
-                super.addViolation(data, node);
-                return data;
-            }
+	        
+	        // Get the instantiated type - We can't use getFirstDescendantOfType since that is a DFS search
```

Comment:
I just realized I could use `getFirstChildOfType`. Will be updating this PR soon.